packages/rna-bundler/lib/index.js

```diff
@@ -9,6 +11,88 @@ export * from './transform.js';
 export * from './build.js';
 export { loadPlugins, loadTransformPlugins, saveManifestJson, saveEntrypointsJson, saveDevEntrypointsJson };
 
+/**
+ * Writes a JSON file with the metafile contents, for bundle analysis.
+ *
+ * @param {import('esbuild').Metafile[]} bundleFiles Array of metafiles for all bundle's generated files
+ * @param {string} filePath Path of the JSON file to generate, relative to CWD
+ * @return {Promise<void>}
+ */
+const writeMetafile = (bundleFiles, filePath) => {
+    const bundle = bundleFiles.reduce((bundle, /** @type {import('esbuild').Metafile} */ metaFile) => {
+        bundle.inputs = { ...bundle.inputs, ...metaFile.inputs };
+        bundle.outputs = { ...bundle.outputs, ...metaFile.outputs };
+
+        return bundle;
+    }, { inputs: {}, outputs: {} });
+
+    return writeFile(filePath, JSON.stringify(bundle))
+        .then((err) => {
+            if (err !== undefined) {
```

Comment:
This is how fs callbacks used to work, but the promise based implementation throws if an error occurred, you should use a try/catch block:

```
try {
  await writeFile(...);
  log();
} catch (err) {
  logError();
}
```